### PR TITLE
[settings] support enum in tree editor

### PR DESCRIPTION
**Pull request by 3nids**
This introduces combo boxes from enum based settings

image

### CODE REVIEW

File: src/gui/CMakeLists.txt [hunk 1]
```diff
@@ -1713,6 +1714,10 @@ target_link_libraries(qgis_gui
   ${QT_VERSION_BASE}::MultimediaWidgets
 )
 
+if (HAVE_ANALYSIS)
```
Comment (nyalldawson) [inline src/gui/CMakeLists.txt:1717]: I'm a big -1 to this -- gui should not be linking to analysis code. What's the reasoning here?

**Comment by 3nids** [comment]
This is ready for review.

@nyalldawson there is no linking to analysis code anymore

**Comment by 3nids** [comment]
test `PyQgsExternalStorageAwsS3` is constantly failing, I don't see this related.

**Comment by m-kuhn** [comment]
Probably https://github.com/qgis/QGIS/pull/54131 ?

**Review by m-kuhn — APPROVED** [review]
LGTM

**Pull request merged**